lib/Model/BlogPost.php

```diff
@@ -42,6 +46,24 @@ public function __construct(
         $this->date        = $date;
     }
 
+    public static function loadMetadata(ClassMetadataInterface $metadata) : void
+    {
+        $metadata->setIdentifier(['url']);
+    }
+
+    /**
+     * @param mixed[] $project
+     */
+    public function hydrate(array $project, ObjectManagerInterface $objectManager) : void
```

Comment:
Types for the properties should be handled here or we can get an invalid object state. Too bad `hydrate()` doesn't expect an Exception.

Reply:
I'm looking forward to typed properties too. The hydrator is currently the part where no wrong types should be able to reach its destined property.